### PR TITLE
PW-6762: Fix HMAC verification functions corrupt notification item payload

### PR DESCRIPTION
**Pull request by alainburindi**
**Description**
Do a copy of the payload to avoid changing the main notification payload
**Tested scenarios**

**Fixed issue**:  #148

### CODE REVIEW

**Comment by alainburindi** [comment]
> Thanks for this contribution @alainburindi. I believe this would break usages of those functions that for some reason are expecting the passed payload to change, right?

Yes, in case there are any. In our use case, we don't want the payload to be changed because our ERP uses one endpoint for different accounts.

We can add another parameter on the function that will be used to check whether the user wants the payloads to be changed or not.

```python
def is_valid_hmac(dict_object, hmac_key, keep_additional_data=False):
    if keep_additional_data:
        dict_object = dict_object.copy()
# ...rest
```

@acampos1916  What do you think? with this, we won't break any existing usages that expect the payload to change

**Comment by acampos1916** [comment]
I like that @alainburindi

What do you think @AlexandrosMor?

**Comment by AlexandrosMor** [comment]
@alainburindi, Thank you for opening this pull request. yes indeed I prefer also that solution

**Comment by alainburindi** [comment]
@acampos1916 @AlexandrosMor I have done the changes accordingly

**Review by acampos1916 — DISMISSED** [review]

File: Adyen/util.py [hunk 1]
```diff
@@ -34,7 +34,10 @@ def escape_val(val):
     return base64.b64encode(hm.digest())
 
 
-def is_valid_hmac(dict_object, hmac_key):
+def is_valid_hmac(dict_object, hmac_key, keep_additional_data=False):
+    if keep_additional_data:
+        dict_object = dict_object.copy()
+
```
Comment (AlexandrosMor) [inline Adyen/util.py:40]:
```suggestion
def is_valid_hmac(dict_object, hmac_key):
        dict_object = dict_object.copy()
```
We decided to always keep the origin additional data.

Comment (alainburindi) [reply Adyen/util.py]: @AlexandrosMor I just modified the changes accordingly. can you check them?

**Review by AlexandrosMor — DISMISSED** [review]

**Pull request merged**